### PR TITLE
Adds framework for simple animals with hands

### PR DESCRIPTION
**Pull request by ghost**
Added framework for simple animals with hands. By default, they have no unique slots other than the left and right hand. They're able to do anything that a human could involving their equipped items. This doesn't work for some simple animals if their code doesn't support it.

The variable used is `dextrous` - if it's 1, then the mob has hands. This can be applied to existing mobs, but the client of the mob has to either reconnect or be forced out of the mob then put back in.

**Token: #18954**

### CODE REVIEW

**Comment by WJohn** [comment]
How does this change things like syndicate mobs which have guns in their hands? Will they use real guns or is it just mysteriously going to fire bullets from nothingness?

**Comment by Xhuis** [comment]
@WJohn There is no change in existing simple animals. They will function the same way.

**Comment by ChangelingRain** [comment]
Plus we can't easily give those hands because they're a /hostile subtype and that'd end up as copypasting probably.

**Comment by MMMiracles** [comment]
Is there any reason why it can't just have a toggle defaulting off and placing it directly for all simple animals instead of making it a sub-type?

I don't exactly see the point of having such a limited mob when a lot of things would heavily benefit from having the option of giving working hands to.

**Comment by Xhuis** [comment]
That's... actually a really good point.

**Comment by RemieRichards** [comment]
@Xhuis can you rip out drone hands? they should just be ```dexterous = TRUE``` now.

**Comment by Xhuis** [comment]
Changes applied to coincide with what @MMMiracles suggested. Any animal can now have and use hands by using the `dextrous` variable.

@RemieRichards Sure, but they have a unique HUD that I'll have to screw around with.

**Comment by RemieRichards** [comment]
@Xhuis their hud would probably just be a subtype of the new dexterous hud.

**Comment by Armhulen** [comment]
When are we giving hivebot five arms with nodrop guns on them?

File: code/_onclick/hud/generic_dextrous.dm [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+	mymob.client.screen = list()
+
+	for(var/obj/screen/inventory/inv in (static_inventory + toggleable_inventory))
+		if(inv.slot_id)
+			inv.hud = src
+			inv_slots[inv.slot_id] = inv
+			inv.update_icon()
+
+/datum/hud/destrous/persistant_inventory_update()
```
Comment (optimumtact) [inline code/_onclick/hud/generic_dextrous.dm]: persistent

destrous ?????

Comment (Xhuis) [inline code/_onclick/hud/generic_dextrous.dm]: MIsspelled "dextrous".
"persistant" is how it is in the code and I can't change it.

**Pull request merged**